### PR TITLE
tests: add tcp split handshake tests

### PR DESCRIPTION
**Pull request by victorjulien**

### CODE REVIEW

**Comment by victorjulien** [comment]
Yes, this was addressed in the code many years ago, and surprisingly it still worked :)

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Based on split handshake research by Tod Beardsley
+
+https://www.macrothink.org/journal/index.php/npa/article/view/285/807
```
Comment (catenacyber) [inline tests/README.md:3]: Should be in a subdirectory

File: tests/tcp-split-handshake-01-4whs/test.rules [hunk 2]
```diff
@@ -0,0 +1,2 @@
+alert tcp any any -> any any (content:"EVIL"; sid:1;)
+alert tcp-stream any any -> any any (content:"EVIL"; sid:2;)
```
Comment (catenacyber) [inline tests/tcp-split-handshake-01-4whs/test.rules:2]: Is there an expected difference in behavior for these 2 sigs ?

Comment (victorjulien) [reply tests/tcp-split-handshake-01-4whs/test.rules:2]: no, added `tcp-pkt` variant to make sure they all match

**Comment by victorjulien** [comment]
replaced by #1959

**Pull request closed without merging**